### DISCUSSION TITLE
Windows Devices refusing connection only working "sometimes"

### DISCUSSION THREAD

**Discussion by DylanTX** [rustdesk/rustdesk-server-pro#492] [Q&A]
Hi Everyone, 

New to RustDesk loving it so far. Only problem is my windows computers will sometimes refuse a connection with a "invalid key error" some times spin on "connecting" endlessly. 1-3min. And other times work fine and connect within 10 seconds. This is all under the same good network conditions I have attempted to connect w and w/o a VPN. Connections work properly 10% of the time. Am I doing something wrong? Any suggested troubleshooting steps? 

Thanks!

**Comment by rustdesk** [comment]
@21pages follow it up

**Comment by rustdesk** [comment]
What Pro server version are you using? Docker or docker compose? what os?
